odoo/models.py

```diff
@@ -3189,24 +3186,28 @@ def _check_company(self, fnames=None):
                 corecord = record.sudo()[name]
                 if corecord._name == 'res.users' and corecord.company_ids:
                     if not (company <= corecord.company_ids):
-                        inconsistent_fields.add(name)
-                        inconsistent_recs |= record
+                        inconsistencies.append((record, name, corecord))
                 elif not (corecord.company_id <= company):
-                    inconsistent_fields.add(name)
-                    inconsistent_recs |= record
-
-        if inconsistent_fields:
-            message = _("""Some records are incompatible with the company of the %(document_descr)s.
-
-Incompatibilities:
-Fields: %(fields)s
-Record ids: %(records)s
-""")
-            raise UserError(message % {
-                'document_descr': self.env['ir.model']._get(self._name).name,
-                'fields': ', '.join(sorted(inconsistent_fields)),
-                'records': ', '.join([str(a) for a in inconsistent_recs.ids[:6]]),
-            })
+                    inconsistencies.append((record, name, corecord))
+
+        if inconsistencies:
+            lines = [_("Incompatible companies on records:")]
+            company_msg = _(" - Record is company %(company)r and %(field)r (%(fname)s: %(values)s) belongs to another company.")
```

Comment:
```suggestion
            company_msg = _("- Record is company %(company)r and %(field)r (%(fname)s: %(values)s) belongs to another company.")
```

Starting and trailing spaces tends to be forgotten/removed/added in translations so it's usually better to avoid them to have a consistent result.